### PR TITLE
Categories

### PR DESCRIPTION
**Pull request by kave**
## Additions

- Added specific Blog/Newsroom Categories to Related Post options

## Changes

- Updated article template to render all categories

## Testing

- visit [here](http://localhost:8000/blog/you-have-the-right-to-someone-in-your-financial-corner/) and see that blogs now render all categories icons

- Load Refresh DB
 - visit [here](http://localhost:8000/foia-requests/) and see it pulls a hodge podge of posts
 - edit this [page](http://localhost:8000/admin/pages/640/edit/) and in the sidebar Related posts organism select a specific category or either blogs or newsrooms or both! Then Publish this page
  -  visit [here](http://localhost:8000/foia-requests/) and see it pulls a posts related to your selections

## Review

- @kurtw 
- @richaagarwal 

## Screenshots
before:
screen shot 2016-03-29 at 5 09 27 pm

edited:
screen shot 2016-03-29 at 9 16 03 pm

after:
screen shot 2016-03-29 at 9 17 06 pm

### CODE REVIEW

**Comment by schaferjh** [comment]
Nice start...some questions:
So that drop down includes both blog and newsroom categories? 
And if I don't make any selection, it'll just pull in items tagged with the relevant topic tags in all categories? 
And if I only make a blog category selection, it'll still pull in all categories of news items?

I have to admit, I'm a little worried by the screenshots, cause it looks like the category selection is overriding the topic tag. 

If that's the FOIA page, then the results of the FOIA topic tag + the Data & Research Category should look like this in the Blog portion:
/blog/?filter_blog_category=Data%2C+Research+%26amp%3B+Reports&filter_tags=FOIA&filter_range-date-gte=&filter_range-date-lte=
But instead, what's in your screen shot looks like this: 
/blog/?filter_blog_category=Data%2C+Research+%26amp%3B+Reports&filter_range-date-gte=&filter_range-date-lte=

**Comment by kave** [comment]
@schaferjh 
1) Yes it contains both blog & newsroom categories
2) If nothing it selected it follows the default behavior of pulling in related posts by tag
3) Yes if you only make a blog category selection, the newsroom will maintain its default behaviour (as seen in the 2nd screenshot)
4) yea, i had coded it to find posts using the logic `FOIA or Data & Research`, but i can rework it make it `FOIA and Data & Research`

**Comment by schaferjh** [comment]
Great re: 1-3, thanks!

> 4) yea, i had coded it to find posts using the logic FOIA or Data & Research, but i can rework it make it FOIA and Data & Research. I went this route because our query looks up posts with tags using the Or logic i.e post is selected if it had TagA or TagB or TagC vs TagA and TagB and TagC

We definitely need to maintain the subject matter focus provided by the tags. The categories aren't topic tags per the filters -- they're another filtering facet, so hopefully we can match the filter behavior of categoryA or categoryB AND topicA or topicB or topicC. 

Certainly, it's an interesting option to just provide a feed of a specific category with no topic tag focus, but that would be a secondary option, not the primary.

**Comment by kave** [comment]
Edited queries to match feedback `categoryA or categoryB AND topicA or topicB or topicC` to better match blog filtering on `/blog/`

blog filter:
screen shot 2016-03-30 at 9 37 52 am

related posts:
screen shot 2016-03-30 at 9 37 57 am

File: cfgov/jinja2/v1/_queries/filtered_17.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+      ]
+    },
+    "query": {
+      "bool": {
+        "should": [
+        ]
+      }
+    }
+  }
+}
```
Comment (kave) [inline cfgov/jinja2/v1/_queries/filtered_17.json]: using ES 1.7 format. Will need to be edited if we ever make an ES upgrade to 2.2

**Comment by kurtw** [comment]
Great job :+1:

**Comment by richaagarwal** [comment]
:+1:

**Pull request merged**